src/chart/graph/categoryVisual.js

```diff
@@ -31,30 +31,53 @@ export default function (ecModel) {
             var name = categoriesData.getName(idx);
             // Add prefix to avoid conflict with Object.prototype.
             categoryNameIdxMap['ec-' + name] = idx;
-
             var itemModel = categoriesData.getItemModel(idx);
+
             var color = itemModel.get('itemStyle.color')
                 || seriesModel.getColorFromPalette(name, paletteScope);
             categoriesData.setItemVisual(idx, 'color', color);
+
+            var opacity = itemModel.get('itemStyle.opacity') || 1;
+            categoriesData.setItemVisual(idx, 'opacity', opacity);
+
+            var symbolStyleList = ['symbolSize', 'symbolKeepAspect'];
+            for (var i = 0; i < symbolStyleList.length; i++) {
+                var symbolStyleItem = itemModel.getShallow(symbolStyleList[i], true);
+                if (symbolStyleItem != null) {
+                    categoriesData.setItemVisual(idx, symbolStyleList[i], symbolStyleItem);
+                }
+            }
+
         });
 
         // Assign category color to visual
         if (categoriesData.count()) {
             data.each(function (idx) {
+                var itemModel = categoriesData.getItemModel(idx);
+                var symbol = itemModel.getShallow('symbol', true);
+                if (symbol) {
+                    data.setItemVisual(idx, 'symbol', symbol);
+                }
+
                 var model = data.getItemModel(idx);
                 var category = model.getShallow('category');
                 if (category != null) {
                     if (typeof category === 'string') {
                         category = categoryNameIdxMap['ec-' + category];
                     }
-                    if (!data.getItemVisual(idx, 'color', true)) {
-                        data.setItemVisual(
-                            idx, 'color',
-                            categoriesData.getItemVisual(category, 'color')
-                        );
+
+                    var itemStyleList = ['color', 'opacity', 'symbolSize', 'symbolKeepAspect'];
+
+                    for (var i = 0; i < itemStyleList.length; i++) {
+                        if (!data.getItemVisual(idx, itemStyleList[i], true)) {
```

Comment:
Should not use `!` in condition here. It is error-prone, for example, consider 0 value.

```js
if (data.getItemVisual(...) == null) { ... }
```
would be more safe.